linux/LinuxProcessTable.c

```diff
@@ -682,21 +682,22 @@ static void LinuxProcessTable_readMaps(LinuxProcess* process, openat_arg_t procF
 }
 
 static bool LinuxProcessTable_readStatmFile(LinuxProcess* process, openat_arg_t procFd, const LinuxMachine* host) {
-   FILE* statmfile = fopenat(procFd, "statm", "r");
-   if (!statmfile)
+   char statmdata[128] = {0};
+
+   if (xReadfileat(procFd, "statm", statmdata, sizeof(statmdata)) < 1) {
       return false;
+   }
 
    long int dummy, dummy2;
 
-   int r = fscanf(statmfile, "%ld %ld %ld %ld %ld %ld %ld",
+   int r = sscanf(statmdata, "%ld %ld %ld %ld %ld %ld %ld",
```

Comment:
I wonder if it is a good idea to skip the two dummy arguments by using the format `"%ld %ld %ld %ld %*ld %ld %*ld"` ?

Reply:
This is kinda unrelated to what this PR is aiming at (further patches incoming).

Also, given this is all stack memory, it's not a memory issue (focus of this PR), but a performance one. And if we were to go for the performance aspect (no plans as of now) we'd still have my `fast_strtoull` implementation here …

Reply:
I'm okay if the `LinuxProcessTable` performance improvements are addresses in a separate commit or PR.
Just to note that there are rooms to improve the code overall.
(BTW, I think I can slightly tweak the `fast_strtoull_hex` so that it uses one less instruction. Mind if I submit a draft for that?)

Reply:
As the `fast_strtoull_hex` is used in some time-critical places, submitting a draft for a performance improvement for it would be good. Please include measurements with valgrind/callgrind or some other performance measurement tool to compare.

Reply:
Yes, the `isxdigit` was written by me. The constant stems from masking `0x1F`, after that, you only care about things `0x00` through `0x1F`, which corresponds to `0x30` through `0x4F`. The bits in that constant are basically `0x00..0x0F` -> hex digits `'A'..'F'` and `0x10..0x1F` -> `'0'..'9'`. Hope this explains that constant.

Reply:
I do understand how the constant works, and it was clever. I try to extend your idea to filter upper bits of a byte, too. `(0x58 & (1L << (ch >> 4)))` Can't say if this would be faster, but I run into some compiler problems that it somehow missed the optimizations that I expected.

Reply:
Mostly noted the construction here for reference, so people less versed in the code can get a feel for what this constant actually represents.